src/node_path.cc

```diff
@@ -0,0 +1,132 @@
+#include "node_path.h"
+#include "env-inl.h"
+#include "node_errors.h"
+#include "node_external_reference.h"
+#include "util-inl.h"
+#include "v8-fast-api-calls.h"
+
+namespace node {
+
+namespace path {
+using v8::Context;
+using v8::FunctionCallbackInfo;
+using v8::Local;
+using v8::Object;
+using v8::Value;
+
+// extracted from
+// https://github.com/torvalds/linux/blob/cdc9718d5e590d6905361800b938b93f2b66818e/lib/glob.c
+bool glob(char const* pat, char const* str) {
+  /*
+   * Backtrack to previous * on mismatch and retry starting one
+   * character later in the string.  Because * matches all characters
+   * (no exception for /), it can be easily proved that there's
+   * never a need to backtrack multiple levels.
+   */
+  char const* back_pat = nullptr;
+  char const* back_str = nullptr;
+
+  /*
+   * Loop over each token (character or class) in pat, matching
+   * it against the remaining unmatched tail of str.  Return false
+   * on mismatch, or true after matching the trailing nul bytes.
+   */
+  for (;;) {
```

Comment:
Nit: There is no need to iterate each character. input.find_first_of("?*[") until the input is finished might produce better results if performance is the priority. If not, we can keep it as it is, and later improve it. Your call.

Reply:
I'm not sure I understand these comments - is there really motivation in diverging from the implementation lifted from `glob` rather than benefit from its updates?

If we have improvement suggestions wouldn't it be better to upstream them to linux glob? 

(genuinely asking, not sure it's. just not really intuitive)